tests/standard cases/gen Wagner/Al+O/1730302779_config.txt
OXIDANTS:
    PRIMARY:
        ELEMENT: O
        DIFFUSION_CONDITION: O in Ni Krupp
        MASS_CONCENTRATION: 0
        CELLS_CONCENTRATION: 0.001
        CONC_PRECISION: rand
        SPACE_FILL: full
        DENSITY: 1
        MOLAR_MASS: 0.016
        DIFFUSION_COEFFICIENT: 2.8231080610996937e-12
        PROBABILITIES: [0.11018779504069587, 0.22037559008139174, 0.3388732297558248]
        PROBABILITIES_2D: 0.9824746403359512
        PROBABILITIES_SCALE: [0.00313238911784397, 0.9812056652929362, 0.00626477823568794]
        PROBABILITIES_INTERFACE: [0.00019687305382588838, 0.0003937461076517979, 0.9988187616770448]
        N_PER_PAGE: 10
        MOLES_PER_CELL: 1.40592582754076e-13
        MASS_PER_CELL: 2.2494813240652158e-15
    SECONDARY:
        ELEMENT: None
        DIFFUSION_CONDITION: None
        MASS_CONCENTRATION: 0
        CELLS_CONCENTRATION: 0
        CONC_PRECISION: rand
        SPACE_FILL: full
        DENSITY: 1
        MOLAR_MASS: 1
        DIFFUSION_COEFFICIENT: 1e-13
        PROBABILITIES: [0.010753022397498174, 0.9354818656150108, 0.021506044794996515]
        PROBABILITIES_2D: 0.014545913237781472
        PROBABILITIES_SCALE: [0.07752111693461146, 0.5348732983923312, 0.155042233869223]
        N_PER_PAGE: 0
        MOLES_PER_CELL: 0.0
        MASS_PER_CELL: 0.0
    SECONDARY_EXISTENCE: False
ACTIVES:
    PRIMARY:
        ELEMENT: Al
        DIFFUSION_CONDITION: Al in Ni Krupp
        MASS_CONCENTRATION: 0.07
        CELLS_CONCENTRATION: 0.2
        CONC_PRECISION: exact
        SPACE_FILL: full
        DENSITY: 2700
        MOLAR_MASS: 0.027
        DIFFUSION_COEFFICIENT: 2.2164389765037816e-14
        ATOMIC_CONCENTRATION: 0.14062767377391427
        N_PER_PAGE: 2081
        PROBABILITIES: [0.12476108824163666, 0.25143347055018006, 0.24952217648327324]
        T: 1.5175468483816013
        n_ELEM: -0.5175468483816013
        MOLES_PER_CELL: 9.372838850271732e-14
        MASS_PER_CELL: 2.5306664895733675e-15
        EQ_MATRIX_MOLES_PER_CELL: 1.42237220576185e-13
        EQ_MATRIX_MASS_PER_CELL: 8.34932484782206e-15
    SECONDARY:
        ELEMENT: None
        DIFFUSION_CONDITION: None
        MASS_CONCENTRATION: 0
        CELLS_CONCENTRATION: 0
        CONC_PRECISION: rand
        SPACE_FILL: full
        DENSITY: 1
        MOLAR_MASS: 1
        DIFFUSION_COEFFICIENT: 1e-13
        ATOMIC_CONCENTRATION: 0.0
        N_PER_PAGE: 0
        PROBABILITIES: [0.045499813632763356, 0.09099962726552674, 0.72700111820342]
        T: 151754.68483816014
        n_ELEM: -151753.68483816014
        MOLES_PER_CELL: 0
        MASS_PER_CELL: 0
        EQ_MATRIX_MOLES_PER_CELL: 0.0
        EQ_MATRIX_MASS_PER_CELL: 0.0
    SECONDARY_EXISTENCE: False
PRODUCTS:
    PRIMARY:
        PHASE_FRACTION_LIMIT: 0
        THRESHOLD_INWARD: 1
        THRESHOLD_OUTWARD: 1
        MASS_PER_CELL: 4.780147813638583e-15
        MOLES_PER_CELL: 4.686419425135866e-14
        MOLES_PER_CELL_TC: 2.343209712567933e-13
        CONSTITUTION: Al+O
        OXIDATION_NUMBER: 1
        LIND_FLAT_ARRAY: 6
    SECONDARY:
        PHASE_FRACTION_LIMIT: 0
        THRESHOLD_INWARD: 0
        THRESHOLD_OUTWARD: 0
        MASS_PER_CELL: 2.2494813240652158e-15
        MOLES_PER_CELL: 0.0
        MOLES_PER_CELL_TC: 0.0
        CONSTITUTION: None+O
    TERNARY:
        PHASE_FRACTION_LIMIT: 0
        THRESHOLD_INWARD: 0
        THRESHOLD_OUTWARD: 0
        MASS_PER_CELL: 2.5306664895733675e-15
        MOLES_PER_CELL: 9.372838850271732e-14
        MOLES_PER_CELL_TC: 1.8745677700543464e-13
        CONSTITUTION: Al+None
    QUATERNARY:
        PHASE_FRACTION_LIMIT: 0
        THRESHOLD_INWARD: 0
        THRESHOLD_OUTWARD: 0
        MASS_PER_CELL: 0.0
        MOLES_PER_CELL: 0
        MOLES_PER_CELL_TC: 0
        CONSTITUTION: None+None
    QUINT:
        PHASE_FRACTION_LIMIT: 0
        THRESHOLD_INWARD: 0
        THRESHOLD_OUTWARD: 0
MAP_PRODUCTS_TO_ELEMENTS: True
MATRIX:
    ELEMENT: Ni
    DIFFUSION_CONDITION: None
    MASS_CONCENTRATION: 0
    CELLS_CONCENTRATION: 0
    CONC_PRECISION: rand
    SPACE_FILL: full
    DENSITY: 8908
    MOLAR_MASS: 0.0587
    MOLES_PER_CELL: 1.4300182889514604e-13
    MASS_PER_CELL: 8.394207356145073e-15
TEMPERATURE: 1100
N_CELLS_PER_AXIS: 102
N_ITERATIONS: 1000000
STRIDE: 100
STRIDE_MULTIPLIER: 50
PRECIP_TRANSFORM_DEPTH: 20
SIM_TIME: 72000
SIZE: 9.999999999999999e-05
SOL_PROD: 6.2500000000000005e-31
PHASE_FRACTION_LIMIT: 1
THRESHOLD_INWARD: 1
THRESHOLD_OUTWARD: 1
NEIGH_RANGE: 1
N_BOOST_STEPS: 1
PROD_INCR_CONST: 1e-05
PROD_ERROR: 0.01
ZETTA_ZERO: 1e-06
ZETTA_FINAL: 4.9999999999999996e-05
INWARD_DIFFUSION: True
OUTWARD_DIFFUSION: True
COMPUTE_PRECIPITATION: True
SAVE_WHOLE: False
DECOMPOSE_PRECIPITATIONS: False
FULL_CELLS: False
SAVE_PATH: C:/test_runs_data/
SAVE_POST_PROCESSED_INPUT: True
MULTIPROCESSING: True
NUMBER_OF_PROCESSES: 24
NUMBER_OF_DIVS_PER_PAGE: 1
DEPTH_PER_DIV: 1
MAX_TASK_PER_CHILD: 50000
BUFF_SIZE_CONST_ELEM: 1.5
TERMINATION_COMMAND: ctrl+g+m
PROBABILITIES:
    PRIMARY:
        p0: None
        p0_f: None
        p0_A_const: None
        p0_B_const: None
        p1: None
        p1_f: None
        p1_A_const: None
        p1_B_const: None
        global_A: None
        global_B: None
        global_B_f: None
        max_neigh_numb: None
        nucl_adapt_function: None
        p0_d: None
        p0_d_f: None
        p0_d_A_const: None
        p0_d_B_const: None
        p1_d: None
        p1_d_f: None
        p1_d_A_const: None
        p1_d_B_const: None
        p6_d: None
        p6_d_f: None
        p6_d_A_const: None
        p6_d_B_const: None
        global_d_A: None
        global_d_B: None
        global_d_B_f: None
        bsf: None
        dissol_adapt_function: None
    SECONDARY:
        p0: None
        p0_f: None
        p0_A_const: None
        p0_B_const: None
        p1: None
        p1_f: None
        p1_A_const: None
        p1_B_const: None
        global_A: None
        global_B: None
        global_B_f: None
        max_neigh_numb: None
        nucl_adapt_function: None
        p0_d: None
        p0_d_f: None
        p0_d_A_const: None
        p0_d_B_const: None
        p1_d: None
        p1_d_f: None
        p1_d_A_const: None
        p1_d_B_const: None
        p6_d: None
        p6_d_f: None
        p6_d_A_const: None
        p6_d_B_const: None
        global_d_A: None
        global_d_B: None
        global_d_B_f: None
        bsf: None
        dissol_adapt_function: None
GENERATED_VALUES:
    TAU: 0.072
    LAMBDA: 9.80392156862745e-07
    KINETIC_KONST: 1.8261221816248283e-07
    inward_moles: 1.4059258275407598e-12
    inward_mass: 2.249481324065216e-14
    sinward_moles: 0.0
    sinward_mass: 0.0
    outward_moles: 1.9504877647415474e-10
    outward_mass: 5.266316964802177e-12
    outward_eq_mat_moles: 2.9599565601904096e-10
    soutward_moles: 0
    soutward_mass: 0
    soutward_eq_mat_moles: 0.0
    matrix_moles: 1.1917953718060583e-09
    matrix_mass: 6.995838832501562e-11
    whole_moles: 1.3882500741077538e-09
    whole_mass: 7.524720010305844e-11
    inward_c_moles: 0.0010127323986957944
    inward_c_mass: 0.00029894551836936523
    sinward_c_moles: 0.0
    sinward_c_mass: 0.0
    outward_c_moles: 0.14049974144572988
    outward_c_mass: 0.069986882669248
    soutward_c_moles: 0.0
    soutward_c_mass: 0.0
    matrix_c_moles: 0.8584875261555743
    matrix_c_mass: 0.9297141718123827
    max_gamma_min_one: 3.2806140788563014e+19
    DB_ID: 1730302779
    DB_PATH: C:/test_runs_data/1730302779.db
    DATE_OF_CREATION: 2024-10-30 16:39:39
COMMENT: This script simulates the general Wagner case with some permeability of active element.
INITIAL_SCRIPT: 
    class NewSystem(SimulationConfigurator):
        def __init__(self):
            super().__init__()
            self.c_automata.cases.first.oxidant.diffuse = self.c_automata.cases.first.oxidant.diffuse_bulk
            self.c_automata.cases.first.active.diffuse = elements.diffuse_bulk_mp

            self.c_automata.get_cur_ioz_bound = self.c_automata.ioz_depth_furthest_inward

            self.c_automata.precip_func = self.c_automata.precipitation_current_case
            self.c_automata.get_combi_ind = self.c_automata.get_combi_ind_standard

            self.c_automata.cases.first_mp.precip_step = precip_step_standard
            self.c_automata.cases.first_mp.check_intersection = ci_single_no_growth

            self.cases.first.fix_init_precip_func_ref = self.c_automata.fix_init_precip_dummy



configuration/config.py
from .config_utils_classes import ElemInput, ElementGroups, ProdInput, ProdGroups, ConfigProbabilities, GeneratedValues


class Config:
    OXIDANTS = ElementGroups()
    OXIDANTS.PRIMARY = ElemInput()
    OXIDANTS.SECONDARY = ElemInput()

    ACTIVES = ElementGroups()
    ACTIVES.PRIMARY = ElemInput()
    ACTIVES.SECONDARY = ElemInput()

    PRODUCTS = ProdGroups()
    PRODUCTS.PRIMARY = ProdInput()
    PRODUCTS.PRIMARY.THRESHOLD_INWARD = 1
    PRODUCTS.PRIMARY.THRESHOLD_OUTWARD = 1

    PRODUCTS.SECONDARY = ProdInput()
    # PRODUCTS.SECONDARY.THRESHOLD_INWARD = 3
    # PRODUCTS.SECONDARY.THRESHOLD_OUTWARD = 2

    PRODUCTS.TERNARY = ProdInput()
    # PRODUCTS.TERNARY.THRESHOLD_INWARD = 4
    # PRODUCTS.TERNARY.THRESHOLD_OUTWARD = 2

    PRODUCTS.QUATERNARY = ProdInput()
    # PRODUCTS.QUATERNARY.THRESHOLD_INWARD = 4
    # PRODUCTS.QUATERNARY.THRESHOLD_OUTWARD = 2

    PRODUCTS.QUINT = ProdInput()
    # PRODUCTS.QUINT.THRESHOLD_INWARD = 1
    # PRODUCTS.QUINT.THRESHOLD_OUTWARD = 0

    MAP_PRODUCTS_TO_ELEMENTS = True

    MATRIX = ElemInput()

    # primary oxidants
    OXIDANTS.PRIMARY.ELEMENT = "O"
    OXIDANTS.PRIMARY.DIFFUSION_CONDITION = "O in Ni Krupp"
    OXIDANTS.PRIMARY.CELLS_CONCENTRATION = 0.01
    # secondary oxidants
    # OXIDANTS.SECONDARY.ELEMENT = "N"
    # OXIDANTS.SECONDARY.DIFFUSION_CONDITION = "N in Ni Krupp"
    # OXIDANTS.SECONDARY.CELLS_CONCENTRATION = 0.01
    # primary actives
    ACTIVES.PRIMARY.ELEMENT = "Al"
    ACTIVES.PRIMARY.DIFFUSION_CONDITION = "Al in Ni Krupp"
    ACTIVES.PRIMARY.MASS_CONCENTRATION = 0.15
    ACTIVES.PRIMARY.CELLS_CONCENTRATION = 0.5
    ACTIVES.PRIMARY.CONC_PRECISION = "rand"
    ACTIVES.PRIMARY.SPACE_FILL = "full"
    # secondary actives
    # ACTIVES.SECONDARY.ELEMENT = "Al"
    # ACTIVES.SECONDARY.DIFFUSION_CONDITION = "Al in Ni Krupp"
    # ACTIVES.SECONDARY.MASS_CONCENTRATION = 0.04
    # ACTIVES.SECONDARY.CELLS_CONCENTRATION = 2
    # 0.0308148148148148000000
    # 0.0770370370370371000000
    # 0.0038518518518518500000
    # 0.0154074074074074000000
    # 0.3851851851851850000000
    # 0.361111111
    # 1.20370370370370000000
    # 1.54074074074074000000
    # 0.92444444444444400000
    # 0.51923076923077
    # 0.96296296296296300000
    # 0.77037037037037000000
    # 10.01481481481480000000

    # matrix
    MATRIX.ELEMENT = "Ni"

    TEMPERATURE = 1100  # °C
    N_CELLS_PER_AXIS = 102  # ONLY MULTIPLES OF 3+(neigh_range-1)*2 ARE ALLOWED
    N_ITERATIONS = 500000  # must be >= n_cells_per_axis
    STRIDE = 100  # n_iterations / stride = n_iterations for outward diffusion
    STRIDE_MULTIPLIER = 50
    PRECIP_TRANSFORM_DEPTH = 20
    SIM_TIME = 72000  # [sek]
    SIZE = 500 * (10 ** -6)  # [m]

    SOL_PROD = 6.25 * 10 ** -31  # 5.621 * 10 ** -10
    PHASE_FRACTION_LIMIT = 1
    THRESHOLD_INWARD = 1
    THRESHOLD_OUTWARD = 1
    NEIGH_RANGE = 1   # neighbouring ranges    1, 2, 3, 4, 5,  6,  7,  8,  9,  10
                      #          and           |  |  |  |  |   |   |   |   |   |
                      # corresponding divisors 3, 5, 7, 9, 11, 13, 15, 17, 19, 21
    N_BOOST_STEPS = 1

    PROD_INCR_CONST = 1 * 10 ** -5
    PROD_ERROR = 0.01
    ZETTA_ZERO = 1 * (10 ** -6)  # [m]
    ZETTA_FINAL = 50 * (10 ** -6)  # [m]

    INWARD_DIFFUSION = True
    OUTWARD_DIFFUSION = True
    COMPUTE_PRECIPITATION = True
    SAVE_WHOLE = True
    DECOMPOSE_PRECIPITATIONS = False
    FULL_CELLS = False
    SAVE_PATH = 'C:/test_runs_data/'
    SAVE_POST_PROCESSED_INPUT = True

    # Execution___________________________________________________________________
    MULTIPROCESSING = True
    NUMBER_OF_PROCESSES = 24
    NUMBER_OF_DIVS_PER_PAGE = 1
    DEPTH_PER_DIV = 1
    MAX_TASK_PER_CHILD = 1000
    BUFF_SIZE_CONST_ELEM = 1.5
    TERMINATION_COMMAND = 'ctrl+g+m'

    # PROBABILITIES_______________________________________________________________
    PROBABILITIES = ElementGroups()
    PROBABILITIES.PRIMARY = ConfigProbabilities()
    PROBABILITIES.SECONDARY = ConfigProbabilities()
    # PROBABILITIES.TERNARY = ConfigProbabilities()
    # PROBABILITIES.QUATERNARY = ConfigProbabilities()
    # PROBABILITIES.QUINT = ConfigProbabilities()

    # nucleation primary___________________________
    PROBABILITIES.PRIMARY.p0 = 0.1
    PROBABILITIES.PRIMARY.p0_f = 1
    PROBABILITIES.PRIMARY.p0_A_const = 1
    PROBABILITIES.PRIMARY.p0_B_const = 1
    PROBABILITIES.PRIMARY.p1 = 1
    PROBABILITIES.PRIMARY.p1_f = 1
    PROBABILITIES.PRIMARY.p1_A_const = 1
    PROBABILITIES.PRIMARY.p1_B_const = 1
    PROBABILITIES.PRIMARY.global_A = 1
    PROBABILITIES.PRIMARY.global_B = None
    PROBABILITIES.PRIMARY.global_B_f = -20
    PROBABILITIES.PRIMARY.max_neigh_numb = None
    PROBABILITIES.PRIMARY.nucl_adapt_function = 5
    # # dissolution primary_________________________
    # PROBABILITIES.PRIMARY.p0_d = 0.01
    # PROBABILITIES.PRIMARY.p0_d_f = 1
    # PROBABILITIES.PRIMARY.p0_d_A_const = 1
    # PROBABILITIES.PRIMARY.p0_d_B_const = 5
    # PROBABILITIES.PRIMARY.p1_d = 0.005
    # PROBABILITIES.PRIMARY.p1_d_f = 1
    # PROBABILITIES.PRIMARY.p1_d_A_const = 1
    # PROBABILITIES.PRIMARY.p1_d_B_const = 10
    # PROBABILITIES.PRIMARY.p6_d = 1e-5
    # PROBABILITIES.PRIMARY.p6_d_f = 0.99
    # PROBABILITIES.PRIMARY.p6_d_A_const = 1
    # PROBABILITIES.PRIMARY.p6_d_B_const = 20
    # PROBABILITIES.PRIMARY.global_d_A = 1
    # PROBABILITIES.PRIMARY.global_d_B = None
    # PROBABILITIES.PRIMARY.global_d_B_f = -0.33
    # PROBABILITIES.PRIMARY.n = 2
    # PROBABILITIES.PRIMARY.bsf = 1
    # PROBABILITIES.PRIMARY.dissol_adapt_function = 3
    # ________________________

    # nucleation SECONDARY
    # PROBABILITIES.SECONDARY.p0 = 0.001
    # PROBABILITIES.SECONDARY.p0_f = 1
    # PROBABILITIES.SECONDARY.p0_A_const = 1
    # PROBABILITIES.SECONDARY.p0_B_const = 1
    # PROBABILITIES.SECONDARY.p1 = 0.01
    # PROBABILITIES.SECONDARY.p1_f = 1
    # PROBABILITIES.SECONDARY.p1_A_const = 1
    # PROBABILITIES.SECONDARY.p1_B_const = 1
    # PROBABILITIES.SECONDARY.global_A = 1
    # PROBABILITIES.SECONDARY.global_B = None
    # PROBABILITIES.SECONDARY.global_B_f = -20
    # PROBABILITIES.SECONDARY.max_neigh_numb = None
    # PROBABILITIES.SECONDARY.nucl_adapt_function = 5
    # # dissolution SECONDARY
    # PROBABILITIES.SECONDARY.p0_d = 0.1
    # PROBABILITIES.SECONDARY.p0_d_f = 1
    # PROBABILITIES.SECONDARY.p0_d_A_const = 1
    # PROBABILITIES.SECONDARY.p0_d_B_const = 1
    # PROBABILITIES.SECONDARY.p1_d = 0.01
    # PROBABILITIES.SECONDARY.p1_d_f = 1
    # PROBABILITIES.SECONDARY.p1_d_A_const = 1
    # PROBABILITIES.SECONDARY.p1_d_B_const = 1
    # PROBABILITIES.SECONDARY.p6_d = 1 * 10 ** -4
    # PROBABILITIES.SECONDARY.p6_d_f = 0.99
    # PROBABILITIES.SECONDARY.p6_d_A_const = 1
    # PROBABILITIES.SECONDARY.p6_d_B_const = 1
    # PROBABILITIES.SECONDARY.global_d_A = 1
    # PROBABILITIES.SECONDARY.global_d_B = None
    # PROBABILITIES.SECONDARY.global_d_B_f = -0.001
    # PROBABILITIES.SECONDARY.n = 2
    # PROBABILITIES.SECONDARY.bsf = 10
    # PROBABILITIES.SECONDARY.dissol_adapt_function = 3
    # # ________________________
    #
    # # nucleation TERNARY
    # PROBABILITIES.TERNARY.p0 = 0.1
    # PROBABILITIES.TERNARY.p0_f = 1
    # PROBABILITIES.TERNARY.p0_A_const = 1
    # PROBABILITIES.TERNARY.p0_B_const = 1
    # PROBABILITIES.TERNARY.p1 = 0.3
    # PROBABILITIES.TERNARY.p1_f = 1
    # PROBABILITIES.TERNARY.p1_A_const = 1
    # PROBABILITIES.TERNARY.p1_B_const = 1
    # PROBABILITIES.TERNARY.global_A = 1
    # PROBABILITIES.TERNARY.global_B = None
    # PROBABILITIES.TERNARY.global_B_f = -20
    # PROBABILITIES.TERNARY.max_neigh_numb = None
    # PROBABILITIES.TERNARY.nucl_adapt_function = 3
    # # dissolution TERNARY
    # PROBABILITIES.TERNARY.p0_d = 1 * 10 ** -3
    # PROBABILITIES.TERNARY.p0_d_f = 1
    # PROBABILITIES.TERNARY.p0_d_A_const = 1
    # PROBABILITIES.TERNARY.p0_d_B_const = 1
    # PROBABILITIES.TERNARY.p1_d = 1 * 10 ** -4
    # PROBABILITIES.TERNARY.p1_d_f = 1
    # PROBABILITIES.TERNARY.p1_d_A_const = 1
    # PROBABILITIES.TERNARY.p1_d_B_const = 1
    # PROBABILITIES.TERNARY.p6_d = 1 * 10 ** -7
    # PROBABILITIES.TERNARY.p6_d_f = 0.99
    # PROBABILITIES.TERNARY.p6_d_A_const = 1
    # PROBABILITIES.TERNARY.p6_d_B_const = 1
    # PROBABILITIES.TERNARY.global_d_A = 1
    # PROBABILITIES.TERNARY.global_d_B = None
    # PROBABILITIES.TERNARY.global_d_B_f = -0.001
    # PROBABILITIES.TERNARY.n = 2
    # PROBABILITIES.TERNARY.bsf = 10
    # PROBABILITIES.TERNARY.dissol_adapt_function = 3
    # # ________________________
    #
    # # nucleation QUATERNARY
    # PROBABILITIES.QUATERNARY.p0 = 0.1
    # PROBABILITIES.QUATERNARY.p0_f = 1
    # PROBABILITIES.QUATERNARY.p0_A_const = 1
    # PROBABILITIES.QUATERNARY.p0_B_const = 1
    # PROBABILITIES.QUATERNARY.p1 = 0.3
    # PROBABILITIES.QUATERNARY.p1_f = 1
    # PROBABILITIES.QUATERNARY.p1_A_const = 1
    # PROBABILITIES.QUATERNARY.p1_B_const = 1
    # PROBABILITIES.QUATERNARY.global_A = 1
    # PROBABILITIES.QUATERNARY.global_B = None
    # PROBABILITIES.QUATERNARY.global_B_f = -20
    # PROBABILITIES.QUATERNARY.max_neigh_numb = None
    # PROBABILITIES.QUATERNARY.nucl_adapt_function = 3
    # # dissolution QUATERNARY
    # PROBABILITIES.QUATERNARY.p0_d = 1 * 10 ** -3
    # PROBABILITIES.QUATERNARY.p0_d_f = 1
    # PROBABILITIES.QUATERNARY.p0_d_A_const = 1
    # PROBABILITIES.QUATERNARY.p0_d_B_const = 1
    # PROBABILITIES.QUATERNARY.p1_d = 1 * 10 ** -4
    # PROBABILITIES.QUATERNARY.p1_d_f = 1
    # PROBABILITIES.QUATERNARY.p1_d_A_const = 1
    # PROBABILITIES.QUATERNARY.p1_d_B_const = 1
    # PROBABILITIES.QUATERNARY.p6_d = 1 * 10 ** -7
    # PROBABILITIES.QUATERNARY.p6_d_f = 0.99
    # PROBABILITIES.QUATERNARY.p6_d_A_const = 1
    # PROBABILITIES.QUATERNARY.p6_d_B_const = 1
    # PROBABILITIES.QUATERNARY.global_d_A = 1
    # PROBABILITIES.QUATERNARY.global_d_B = None
    # PROBABILITIES.QUATERNARY.global_d_B_f = -0.001
    # PROBABILITIES.QUATERNARY.n = 2
    # PROBABILITIES.QUATERNARY.bsf = 10
    # PROBABILITIES.QUATERNARY.dissol_adapt_function = 3
    # # ________________________
    #
    # # nucleation QUINT
    # PROBABILITIES.QUINT.p0 = 0.1
    # PROBABILITIES.QUINT.p0_f = 1
    # PROBABILITIES.QUINT.p0_A_const = 1
    # PROBABILITIES.QUINT.p0_B_const = 1
    # PROBABILITIES.QUINT.p1 = 0.3
    # PROBABILITIES.QUINT.p1_f = 1
    # PROBABILITIES.QUINT.p1_A_const = 1
    # PROBABILITIES.QUINT.p1_B_const = 1
    # PROBABILITIES.QUINT.global_A = 1
    # PROBABILITIES.QUINT.global_B = None
    # PROBABILITIES.QUINT.global_B_f = -20
    # PROBABILITIES.QUINT.max_neigh_numb = None
    # PROBABILITIES.QUINT.nucl_adapt_function = 3
    # # dissolution QUINT
    # PROBABILITIES.QUINT.p0_d = 1 * 10 ** -3
    # PROBABILITIES.QUINT.p0_d_f = 1
    # PROBABILITIES.QUINT.p0_d_A_const = 1
    # PROBABILITIES.QUINT.p0_d_B_const = 1
    # PROBABILITIES.QUINT.p1_d = 1 * 10 ** -4
    # PROBABILITIES.QUINT.p1_d_f = 1
    # PROBABILITIES.QUINT.p1_d_A_const = 1
    # PROBABILITIES.QUINT.p1_d_B_const = 1
    # PROBABILITIES.QUINT.p6_d = 1 * 10 ** -7
    # PROBABILITIES.QUINT.p6_d_f = 0.99
    # PROBABILITIES.QUINT.p6_d_A_const = 1
    # PROBABILITIES.QUINT.p6_d_B_const = 1
    # PROBABILITIES.QUINT.global_d_A = 1
    # PROBABILITIES.QUINT.global_d_B = None
    # PROBABILITIES.QUINT.global_d_B_f = -0.001
    # PROBABILITIES.QUINT.n = 2
    # PROBABILITIES.QUINT.bsf = 10
    # PROBABILITIES.QUINT.dissol_adapt_function = 3
    # # ________________________

    GENERATED_VALUES = GeneratedValues()
    COMMENT = """NO COMMENTS"""
    INITIAL_SCRIPT = "\n"
